fix: HTTP headers should be URI escaped

fumin opened the pull request:
This is needed for reports containing Chinese characters.

Without this fix, reports with Chinese names would not load

fumin committed:
HTTP headers should be URI escaped

This is needed for reports containing Chinese characters.

surajshetty3416 commented:
@fumin 👍 for your first contribution.

Chinese characters are not allowed in the doctype name. Would like to know how are you able to replicate this issue?

<img width="1235" alt="Screenshot 2020-04-04 at 11 17 34 AM" src="https://user-images.githubusercontent.com/13928957/78419716-2eac3180-7666-11ea-90ad-82abd074d671.png">

fumin commented:
Hi Suraj, thanks for the work on frappe, too. We ran into this issue with a "Script Report" with a Chinese name. The Doctype of the report remains English.

The below screenshot shows the Chinese script report in the report list
![Screen Shot 2020-04-04 at 3 07 43 PM](https://user-images.githubusercontent.com/765222/78420937-6aef8a00-7686-11ea-8e2c-37c7a6cac108.png)

This screenshot shows the settings of the Chinese report
![Screen Shot 2020-04-04 at 3 07 50 PM](https://user-images.githubusercontent.com/765222/78420948-7e9af080-7686-11ea-93b9-ba2f060cc2f2.png)

This screenshot shows the javascript error that we are trying to fix
![Screen Shot 2020-04-04 at 3 05 52 PM](https://user-images.githubusercontent.com/765222/78420962-92465700-7686-11ea-911e-7a1f8ddf7dfa.png)

surajshetty3416 committed:
refactor: Remove unnecessary code in query_report

get_user_settings is useless after following refactor
https://github.com/frappe/frappe/pull/5473
https://github.com/frappe/frappe/pull/5834

https://github.com/frappe/frappe/pull/5834/commits/53193d313052d5c935bd8e1861c858c8533e97b1

surajshetty3416 commented:
@fumin Thanks for the explanation.

Even though your change fixes the issue, the real issue was the left-out unwanted code in query_report.js which I've removed in this [commit](https://github.com/frappe/frappe/pull/9850/commits/f51c975759d58f42e81f8b33df94b6d6860e5c78)

Keeping your change since it is adding extra defense.

surajshetty3416 commented:
@fumin Port this to version-12-hotfix as well

fumin commented:
Thanks Suraj for the fix and review : )
